remove Wx dependency & build on Windows

cbielow opened the pull request:
I was looking for a tool to fix my broken MKV (VLC would play it, but Handbrake would not convert it) -- and meteorite did the job.

To get it to build on **Windows** I removed the unnecessary wx-includes in meteorite.h -- I hope the GUI will still build, but it should because the includes are also present in other headers (not tested though -- I don't have wxWidgets installed).

I used the Windows g++ that comes with R-win (gcc 4.6.) and meteorite-cli.exe builds just fine using

> make cli

now (I fixed the inner workings of the makefile; it should run on any gnu make though -- also not tested -- no Linux around atm).

Hope this helps.

cheers
Chris

cbielow committed:
[FIX] removed unnecessary wx-includes for the CLI version

cbielow committed:
[FIX] makefile produces consistent executables on all platforms; wxWidgets is not required for CLI executable any longer;
[DOC] added Windows build instructions / fixed README

PureOcean commented:
Dear **_Chris_**,

I thank you very much for your removed unneccessary wx line from **meteorite-cli.cpp** code. 

Although I did try very hard, I hadn’t been compiling to windows binary cli-code of Andrew. Now, I tried to your fixed the code on many GCC builds for compile. Finally, the meteorite-cli.exe is compiled. But the all compilers given some errors. 

Just like, Meteorite.exe 0.11 (Erdem's last version only supported drag-and-drop), the my compiled **meteorite-cli.exe** stucked at **98%** when repairing to my incompleted/corrupt MKV file.

Most probably, Andrew's improvements don't work on this stuck issues. 

But there ise a workaround. Need "damaged.mkv" file re-mux with **FFMpeg** before Meteorite repaired:

`ffmpeg-3.1.5.exe -i "damaged.mkv" -c:v copy -c:a copy -hide_banner -loglevel panic "fixed_ffmpeg.mkv"`

And then: 

`Meteorite-cli.exe "fixed_ffmpeg.mkv"`

Now, Meteorite-cli.exe can repair the damaged "fixed_ffmpeg.mkv" without stuck at %98. Just like the MeteoriteGUI 0.11.

Meanwhile, Need version **3.1.5** of FFmpeg. Later versions of FFmpeg 3.1.5 don't work this workaround. They broke or changed something since version 3.1.5. I don't know why.

_My first request_: Could you please share me your compiled meteorite-cli.exe file? I want to try for confirm.

*

_My second request_:

As you know, Meteorite only create with as added the prefix "_Meteorite._OriginalFileName.mkv" output -repaired- file.

Can you adding support a another output filename? Like this:

`meteorite-cli.exe damaged.mkv fixed.mkv`

*

_My third request_:

Meteorite-cli.exe haven't quiet/silent mode. I tried this:

`Meteorite-cli.exe "damaged.mkv" >nul`

But in this suffix, it's only show to percentages when progressing.

Can you add a silent/quiet mode, too? For example:

`meteorite-cli.exe damaged.mkv fixed.mkv /silent`

*
I'm sorry my bad English grammar.
I am also grateful to you for reading this my message.

My warm regards...

BartMassey committed:
fixed C++ standards rot

Code would not compile because of a couple of changes to the C++
standard since it was last maintained. Fixed.

cbielow committed:
Merge pull request #1 from BartMassey/master

fixed C++ standards rot

mirh commented:
[This](https://github.com/user-attachments/files/19091365/meteorite-master.zip) is the cli compiled with gcc 9.3.
